compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -700,44 +700,43 @@ execMigrate projectOpts opts0 inFile1_ inFile2_ mbDir = do
                         ]
                     pure (src, iuid, depends pkgInfo)
         let unitIdsTopoSorted = reverse $ topSort depGraph
-        createDirectoryIfMissing True genDir
         projectPkgDb <- makeAbsolute projectPackageDatabase
-        forM_ unitIdsTopoSorted $ \vertex -> withCurrentDirectory genDir $ do
+        forM_ unitIdsTopoSorted $ \vertex -> do
             let (src, iuid, _) = vertexToNode vertex
-            unless
+            let iuidString = installedUnitIdString iuid
+            let workDir = genDir </> iuidString
+            createDirectoryIfMissing True workDir
+            putStrLn $ "created workdir: " <> workDir
+            withCurrentDirectory workDir $ do
```

Comment:
Can we add a comment here on why we need to change the directory? It isn’t clear to me why that is required.

Reply:
so that you get the right filepath's for the generated interface files.